os/drivers/wireless/virtual/vdev_handler.c

```diff
@@ -29,31 +29,32 @@
 #include <tinyara/net/if/wifi.h>
 #include "vdev_handler.h"
 
-extern wifi_utils_scan_list_s *vwifi_get_scan_list(void);
+extern trwifi_scan_list_s *vwifi_get_scan_list(void);
 extern struct vwifi_ops *get_vdev_onoff(void);
 extern struct vwifi_ops *get_vdev_auto(void);
 
 static void vwifi_callback_handler(lwnl_cb_status evt)
 {
-	if (evt == LWNL_SCAN_DONE) {
-		wifi_utils_scan_list_s *scanlist = vwifi_get_scan_list();
-		lwnl_postmsg(evt, scanlist);
+	if (evt.evt == LWNL_SCAN_DONE) {
+		trwifi_scan_list_s *scanlist = vwifi_get_scan_list();
+		LWNL_POST_WIFISCANMSG(LWNL_SCAN_DONE, scanlist);
 	} else {
-		lwnl_postmsg(evt, NULL);
+		LWNL_POST_WIFIMSG(evt.evt, NULL, 0);
 	}
 }
 
 static void _generate_disconnect(int argc, char *argv[])
 {
```

Comment:
I think we need to check `argc` for safety.
If not, a fault can be occurred because of invalid access when invalid arguments are passed.
Could you please check and add handling arguments?
```
if (argc <= XX) {
   return;
}
```

Reply:
Only one function calls this function and it's static so adding exception handling code is not necessary.